src/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/PerformanceCounterLib.cs

```diff
@@ -674,7 +678,10 @@ internal bool FindCustomCategory(string category, out PerformanceCounterCategory
                                 // In this case we return an 'Unknown' category type and 'false' to indicate the category is *not* custom.
                                 //
                                 categoryType = PerformanceCounterCategoryType.Unknown;
-                                _customCategoryTable[category] = categoryType;
+                                lock (_customCategoryTableLock)
```

Comment:
You shouldn't need a separate _customCategoryTableLock... just lock on table.

Reply:
you'are right. Thanks!

Reply:
should be ok now